homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -663,17 +826,45 @@ class PipelineInput:
     async def execute(self) -> None:
         """Run pipeline."""
         self.run.start()
-        current_stage = self.run.start_stage
+        current_stage: PipelineStage | None = self.run.start_stage
+        audio_buffer: list[bytes] = []
 
         try:
+            if current_stage == PipelineStage.WAKE_WORD:
+                assert self.stt_stream is not None
+                detect_result = await self.run.wake_word_detection(
+                    self.stt_stream, audio_buffer
+                )
+                if detect_result is None:
+                    # No wake word. Abort the rest of the pipeline.
+                    self.run.end()
+                    return
+
+                current_stage = PipelineStage.STT
+
             # speech-to-text
             intent_input = self.intent_input
             if current_stage == PipelineStage.STT:
                 assert self.stt_metadata is not None
                 assert self.stt_stream is not None
+
+                if audio_buffer:
+
+                    async def buffered_stream() -> AsyncGenerator[bytes, None]:
```

Comment:
Isn’t this more or less similar to `from itertools import chain`.

Reply:
It is except for the async part, which I don't think they've added to itertools yet.